tfs/src/main/java/hudson/plugins/tfs/rm/ReleaseWebHookAction.java

```diff
@@ -0,0 +1,160 @@
+        ReleaseWebHook webHook = null;
+        for (ReleaseWebHookReference webHookName : webHookNames) {
+            if (nameToWebHookMap.containsKey(webHookName.getWebHookName())) {
+                try {
+                    webHook = nameToWebHookMap.get(webHookName.getWebHookName());
+                    logger.fine(String.format("Sending payload event to %s", webHook.getPayloadUrl()));
+                    ReleaseWebHookStatus status = sendJobCompletedEvent(json, webHook);
+
+                    webHookStatus.add(status);
+                } catch (Exception ex) {
+                    logger.log(Level.SEVERE, StringUtils.EMPTY, ex);
+                    webHookStatus.add(new ReleaseWebHookStatus(webHook.getPayloadUrl(), HttpURLConnection.HTTP_INTERNAL_ERROR, ex.toString()));
+                }
+            }
+        }
+
+        build.addAction(new ReleaseWebHookSummaryAction((webHookStatus)));
+        return true;
+    }
+
+    private ReleaseWebHookStatus sendJobCompletedEvent(final JSONObject json, final ReleaseWebHook webHook) throws IOException, NoSuchAlgorithmException, InvalidKeyException {
+        HttpClient client = HttpClientBuilder.create().build();
+        final HttpPost request = new HttpPost(webHook.getPayloadUrl());
+        final String payload = json.toString();
+
+        request.addHeader("Content-Type", "application/json");
+        request.addHeader("Accept", "application/json; api-version=" + apiVersion);
+
+        if (!StringUtils.isEmpty(webHook.getSecret())) {
+            String signature = ReleaseWebHookHelper.getPayloadSignature(webHook.getSecret(), payload);
+            request.addHeader("X-Jenkins-Signature", signature);
+        }
+
+        request.setEntity(new StringEntity(payload));
+        final HttpResponse response = client.execute(request);
+        final int statusCode = response.getStatusLine().getStatusCode();
+
+        ReleaseWebHookStatus status = null;
+        if (statusCode == HttpURLConnection.HTTP_OK) {
+            logger.log(Level.INFO, "sent event payload successfully");
+            status = new ReleaseWebHookStatus(webHook.getPayloadUrl(), statusCode);
+        } else {
+            HttpEntity entity = response.getEntity();
+            String content = EntityUtils.toString(entity);
+            logger.log(Level.WARNING, "Cannot send the event to webhook. Content:" + content);
+            status = new ReleaseWebHookStatus(webHook.getPayloadUrl(), statusCode, content);
+        }
+
+        return status;
+    }
+
+    /**
+     * Implementation of DescriptorImpl.
+     */
+    @Extension
+    public static final class DescriptorImpl extends BuildStepDescriptor<Publisher> {
+
+        @Override
+        public boolean isApplicable(final Class<? extends AbstractProject> jobType) {
+            return true;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Azure DevOps Services Release Webhook";
```

Comment:
pl. rename this back to TFS/Team Services Release Webhook, let's do this name refresh afterwards..

Reply:
done.